src/posix/platform/udp.cpp

```diff
@@ -402,19 +436,22 @@ void platformUdpInit(const char *aIfName)
         DieNow(OT_EXIT_INVALID_ARGUMENTS);
     }
 
-    sPlatNetifIndex = if_nametoindex(aIfName);
-
-    if (sPlatNetifIndex == 0)
+    if (aIfName != gNetifName)
     {
-        perror("if_nametoindex");
+        VerifyOrDie(strlen(aIfName) < sizeof(gNetifName) - 1, OT_EXIT_INVALID_ARGUMENTS);
+        strcpy(gNetifName, aIfName);
```

Comment:
nit: just to be safer
```suggestion
        assert(gNetifIndex == 0);
        strcpy(gNetifName, aIfName);
```